src/modV.js

```diff
@@ -258,6 +259,49 @@ class ModV extends EventEmitter2 {
 
 		return layerIndex;
 	}
+
+	/** @return {ModV.WorkersDataType} */
+	createWorkers() {
+		const palette = new PaletteWorker();
+		palette.on(PaletteWorker.EventType.PALETTE_ADDED, this.paletteAddHandler.bind(this));
+		palette.on(PaletteWorker.EventType.PALETTE_UPDATED, this.paletteUpdateHandler.bind(this));
+
+		return {
+			palette,
+		};
+	}
+
+	/**
+	 * @protected
+	 * @param {string} data
+	 */
+	paletteAddHandler(id) {
+		if(!this.palettes.has(id)) {
+			this.palettes.set(id, {});
+		} else {
+			console.error('Palette with ID', id, 'already exists');
+		}
+	}
+
+	/**
+	 * @protected
+	 * @param {string} id
+	 * @param {*} currentColor
+	 * @param {*} currentStep
+	 * @todo Types
+	 */
+	paletteUpdateHandler(id, currentColor, currentStep) {
+		this.palettes.set(id, {
+			currentColor: currentColor,
+			currentStep: currentStep
+		});
+	}
+
+	/** @param {string} id */
+	removePalette(id) {
```

Comment:
Great, sounds like a very good approach. I'll approve.

I think we should discuss some sort of roadmap on Slack very soon. I'd like to get 1.5 wrapped up maybe by June.